akka-actor-typed-tests/src/test/scala/akka/actor/typed/WatchSpec.scala

```diff
@@ -66,5 +74,46 @@ class WatchSpec extends TestKit("WordSpec")
 
       receivedTerminationSignal.future.futureValue shouldEqual CustomTerminationMessage
     }
+
+    class ErrorTestSetup {
+      val terminator = systemActor(terminatorBehavior)
+      val receivedTerminationSignal: Promise[Message] = Promise()
+
+      val watcher = systemActor(Behaviors.immutable[Message] {
+        case (ctx, StartWatchingWith(watchee, msg)) ⇒
+          ctx.watchWith(watchee, msg)
+          Behaviors.same
+        case (ctx, StartWatching(watchee)) ⇒
+          ctx.watch(watchee)
+          Behaviors.same
+        case (_, msg) ⇒
+          receivedTerminationSignal.success(msg)
+          Behaviors.stopped
+      })
+    }
+    "warn when watch is used after watchWith on same subject" in new ErrorTestSetup {
+      watcher ! StartWatchingWith(terminator, CustomTerminationMessage)
+
+      EventFilter[IllegalStateException](pattern = ".*termination message was not overwritten.*", occurrences = 1) intercept {
+        watcher ! StartWatching(terminator)
+      }
+      terminator ! Stop
+    }
+
+    "warn when watchWitch is used after watchWith with different termination message" in new ErrorTestSetup {
+      watcher ! StartWatchingWith(terminator, CustomTerminationMessage)
+
+      EventFilter[IllegalStateException](pattern = ".*termination message was not overwritten.*", occurrences = 1) intercept {
+        watcher ! StartWatchingWith(terminator, CustomTerminationMessage2)
+      }
+    }
+    "warn when watchWith is used after watch on same subject" in new ErrorTestSetup {
+      watcher ! StartWatching(terminator)
+
+      EventFilter[IllegalStateException](pattern = ".*termination message was not overwritten.*", occurrences = 1) intercept {
+        watcher ! StartWatchingWith(terminator, CustomTerminationMessage)
+      }
+      terminator ! Stop
+    }
```

Comment:
None of the test cases actually looks at the `recevedTerminationSignal`?

Reply:
All the actors die because of the exception, so not so interesting here. I might even remove the `Stop`?

Reply:
Then remove the `receivedTerminationSignal` from `ErrorTestSetup`? Or am I missing that it is used somewhere?